recipe/meta.yaml

```diff
@@ -124,6 +125,20 @@ outputs:
       commands:
         - python -c "import ray; ray.init()"
 
+  - name: ray-default
```

Comment:
what's `ray-default` and how is it different from `ray-core`?

Reply:
We're trying to reduce the dependency footprint, so we moved a number of optional dependencies into `ray[default]` for PyPI installs. I'd like to keep these in sync in conda/pip so that users can rely on one naming convention.

`ray` will only include critical dependencies. `ray[default]` will include dependencies e.g. for the dashboard and autoscaler. The rest of the packages will stay as they are.

Reply:
Cool, nice to know. And much more nice is to have someone who knows the insides! 👍

Reply:
That said, maybe `ray` should be renamed to `ray[minimal]` (or even `ray[core]`, so we have a match between conda-forge and pypi)?..

Reply:
Hey @krfricke, great to have you on board! 

conda-forge doesn't have an installable package called `ray` (because the bioconda channel got that name first...), so the "default" here is called `ray-core`. I would propose to incorporate the changes there.

I also don't mind too much about an additional output if you want to unify the naming (I had [asked](https://github.com/conda-forge/staged-recipes/pull/11160#issuecomment-605724196) about the naming before this feedstock was created; at the time we went with `-core` rather than `-default`...).

What we can also do is move everything to `ray-default`, and keep `ray-core` as a compatibility output for a year and then drop it.

Reply:
So the way it works in pypi is that `ray` (which will be `ray-core` in conda) contains only critical dependencies and is used for external libraries that depend on core ray APIs. `ray[default]` will then contain additional dependencies (e.g. for autoscaler, dashboard), so this will map to the (new) `ray-default` then. So I would propose to keep both `ray-core` and `ray-default` like this - or do you suggest we have `ray-core` mirror `ray-default` in conda for backwards compatibility reasons?

Reply:
I don't think there should be backwards compat here - too little gain for the hassle IMHO.
That said, why `ray-default` (and `ray[default]`) instead of `ray[full]` or `ray[all]`?..

Reply:
> @krfricke `ray[default]` will then contain additional dependencies (e.g. for autoscaler, dashboard),

Similar question as @vnlitvinov - will this contain _all_ dependencies? In that case, I would really go with `ray[all]` as well.

If OTOH `ray[default]` is in the middle somewhere - i.e. we have (in terms of dependencies) `ray < ray[default] < ray[all]`) - then we will need to keep the distinction between `ray-core` and `ray-default`, because other libraries in conda-forge may want to only depend on the core APIs themselves.

> @vnlitvinov: I don't think there should be backwards compat here - too little gain for the hassle IMHO.

An additional output that directly maps to another one is effectively zero cost, both in terms of maintenance & infra. See for example [here](https://github.com/conda-forge/lapack-feedstock/blob/master/recipe/meta.yaml#L245-L251)

Reply:
So yes, `ray[default]` is somewhere in the middle. `ray[all]` includes all dependencies (for rllib, tune, serve), where `ray[default]` only includes those for more basic ray operations (autoscaler, dashboard), and `ray` just the core APIs.

The mental model of `ray < ray[default] < ray[all]` is thus right.

Reply:
> An additional output that directly maps to another one is effectively zero cost, both in terms of maintenance & infra. See for example [here](https://github.com/conda-forge/lapack-feedstock/blob/master/recipe/meta.yaml#L245-L251)

That is true as long as it doesn't introduce more naming conflicts in terms of how things are named on different platforms. I would rather we keep pypi and conda-forge namings aligned as much as possible.

That said, `ray-core {conda-forge}` was an equivalent of `ray {pypi}` when there was no `ray[default] {pypi}`, so I propose we keep it equivalent to just `ray {pypi}` and add a `ray-default {conda-forge}` just like is done here. After that we can think about adding `ray-all` and stuff if needed.

Reply:
Yeah, I wouldn't mismatch names where we can avoid it (which was the guiding principle in naming the outputs [originally](https://github.com/conda-forge/staged-recipes/pull/11160#issuecomment-605695760), which I obviously think we should keep; only `ray` itself couldn't be taken directly).

We should match names and intent of the PyPI packages. The only compatibility output would be for old conda-forge packages, but in this case, it sounds like `ray[default]` will be something new anyway.

Reply:
I think we reached consensus here, so I kept the new naming convention in #30